polkadot/node/core/pvf/src/host.rs

```diff
@@ -873,28 +880,102 @@ fn pulse_every(interval: std::time::Duration) -> impl futures::Stream<Item = ()>
 	.map(|_| ())
 }
 
+/// Check if we can sandbox the root and emit a warning if not.
+///
+/// We do this check by spawning a new process and trying to sandbox it. The process must be
+/// single-threaded, so we can't just fork here. To get as close as possible to running the check in
```

Comment:
Oh I remember, here is the background for this comment:

https://github.com/paritytech/polkadot/pull/7337/#discussion_r1262447125

The comment was not very good, sometimes I leave a lot of context in my head. 😛

Reply:
Doesn't change the fact that we could `fork` instead right? Would be cleaner I think...

Reply:
Or maybe it's better to leave the check in the child worker. `fork(2)`:

          The child process is created with a single thread—the one that
          called fork().  The entire virtual address space of the parent
          is replicated in the child, including the states of mutexes,
          condition variables, and other pthreads objects; the use of
          [pthread_atfork(3)](https://www.man7.org/linux/man-pages/man3/pthread_atfork.3.html) may be helpful for dealing with problems
          that this can cause.

Reply:
I think we could still manually `fork`, but I don't think we should. It requires lower-level and unsafe code. The current code is fine, just the comment is misleading.